docs/manual/common/concepts/ServiceDiscovery.md

```diff
@@ -0,0 +1,39 @@
+
+## Client-Side Service Discovery
+
+From Bonér's [Reactive Microservices Architecture: Design Principles for Distributed Systems](http://www.oreilly.com/programming/free/reactive-microservices-architecture.html) 
+
+> Once the information about each service has been stored it can be made available through a Service Registry that services can use to look the information up—using a pattern called Client-Side Service Discovery.
+
+Lagom creates service clients for each Service Descriptor so that applications can interact with Lagom Services. So [[an app|IntegratingNonLagom]] that wanted to consume the hello service can use the Welcome Service Client and simply invoke the `hello` method. The Welcome Service Client will use the Service Registry to find a valid URL where  `welcome` is available and with that information the request is then fulfilled. This approach requires using Lagom provided code.
+
+When running your code in production the Service Locator plugged into your services will be an element participating in this Client-Side Discovery.
+
+## Server-Side Service Discovery
+
+From Bonér's [Reactive Microservices Architecture: Design Principles for Distributed Systems](http://www.oreilly.com/programming/free/reactive-microservices-architecture.html) 
+
+> Another strategy is to have the information stored and maintained in a load balancer [...] using a pattern called Server-Side Service Discovery.
+
+Lagom Developer Mode starts all your services plus a Service Registry and a Service Gateway. The Service Gateway allows Service Clients to consume the endpoints provided by the Service Descriptors registered. A browser can display a hello message to a user by requesting the `/hello/steve` path to the Service Gateway. The Service Gateway will know what service provide that and will request the Service Registry for an instance of that service. The Service Registry will respond with the host and port of the instance where the request can be fulfilled. Finally the Service Gateway will perform the request and return the result to the browser. In this model, the browser only needs to know where the Service Gateway is. When using server-side discovery, a call on a Service Descriptor can only be reached if its call is added to the ACL.
+
+This section uses HTTP-REST for all the examples but the concepts apply to any kind of traffic be it HTTP, binary over tcp, etc...
+
+Server-Side Service Discovery is very convenient when you can't embed your Service Locator on every client.
+
+## Locatable Services
+
+Lagom also supports disabling locatability of a Service Descriptor. It is a common practice to include infrastructure features in each microservice. Those infrastructure features are published via a Service Descriptor. Despite that, we usually don't want the infrastructure features to be locatable, in that case we can simply setup the Service Descriptor as non locatable which will tell the registrator to ignore it.
```

Comment:
I guess this ties back to the thread in the other PR, but why don't we want infrastructure features to be locatable?